provisioning/roles/golang/tasks/main.yml

```diff
@@ -0,0 +1,10 @@
+---
+- name: create destination for source
+  command: mkdir -p {{ golang_tarball_location }} creates={{ golang_tarball_location }}
+
+- name: copy golang tarball
+  copy: src={{ golang_tarball }} dest={{ golang_tarball_location }}
+
+- name: install go
+  unarchive: src={{ golang_tarball }} dest=/usr/local copy=no
```

Comment:
`{{ golang_tarball }}` needs to be replaced with a variable that is the full path to the tarball on the remote system.